addons/hr_timesheet/models/project.py

```diff
@@ -327,6 +327,32 @@ def _compute_subtask_effective_hours(self):
         for task in self.with_context(active_test=False):
             task.subtask_effective_hours = sum(child_task.effective_hours + child_task.subtask_effective_hours for child_task in task.child_ids)
 
+    def get_group_pattern(self):
+        group_pattern = super().get_group_pattern()
+        group_pattern['planned_hours'] = r'\s(\d+(?:\.\d+)?)[hH]'
+        return group_pattern
+
+    def get_groups_patterns(self):
+        patterns = ['(?:%s)*' % self.get_group_pattern()['planned_hours']]
+        patterns.extend(super().get_groups_patterns())
+        return patterns
+
+    def get_cannot_start_with_patterns(self):
+        patterns = super().get_cannot_start_with_patterns()
+        patterns.extend([r'(?!\d+(?:\.\d+)?(?:h|H))'])
+        return patterns
+
+    def extract_planned_hours(self):
+        planned_hours_group = self.get_group_pattern()['planned_hours']
+        if self.allow_timesheets:
+            self.planned_hours = sum(float(num) for num in re.findall(planned_hours_group, self.display_name))
```

Comment:
Summing sounds weird to me (as much as using h|H, which I thought we decided not to).
But that is a functional concern.